cpp/src/replace/replace.cu

```diff
@@ -0,0 +1,158 @@
+   column_device_view in;
+   T __device__ operator()(size_type i)
+   {
+      auto e = in.element<T>(i);
+      if (isnan(e)) {
+         return std::numeric_limits<T>::quiet_NaN();
+      }
+      if (T{0.0} == e) {
+         return T{0.0};
+      }
+      return e;
+   }
+};
+   
+/* --------------------------------------------------------------------------*/
+/**
+* @brief Functor called by the `type_dispatcher` in order to invoke and instantiate
+*        `normalize_nans_and_zeros` with the appropriate data types.
+*/
+/* ----------------------------------------------------------------------------*/
+struct normalize_nans_and_zeros_kernel_forwarder {
+   // floats and doubles. what we really care about.
+   template <typename T, std::enable_if_t<std::is_floating_point<T>::value>* = nullptr>
+   void operator()(  column_device_view in,
+                     mutable_column_device_view out,
+                     cudaStream_t stream)
+   {      
+      thrust::transform(rmm::exec_policy(stream)->on(stream),
+                        thrust::make_counting_iterator(0),
+                        thrust::make_counting_iterator(in.size()),
+                        out.head<T>(), normalize_nans_and_zeros_lambda<T>{in});      
+   }
+
+   // if we get in here for anything but a float or double, that's a problem.
+   template <typename T, std::enable_if_t<not std::is_floating_point<T>::value>* = nullptr>
+   void operator()(  column_device_view in,
+                     mutable_column_device_view out,
+                     cudaStream_t stream)
+   {
+      CUDF_FAIL("Unexpected non floating-point type.");      
+   }   
+};
+
+} // end anonymous namespace
+
+namespace cudf {
+namespace detail {                                               
+
+void normalize_nans_and_zeros(mutable_column_view in_out,
+                              cudaStream_t stream = 0)
+{   
+   CUDF_EXPECTS(in_out.head() != nullptr, "Null input data");
+   if(in_out.size() == 0 || in_out.head() == nullptr){
+      return;
+   }
+   CUDF_EXPECTS(in_out.type() == data_type(FLOAT32) || in_out.type() == data_type(FLOAT64), "Expects float or double input");
+
+   // wrapping the in_out data in a column_view so we can call the same lower level code.
+   // that we use for the non in-place version.
+   column_view input = in_out;
+
+   // to device. unique_ptr which gets automatically cleaned up when we leave
+   auto device_in = column_device_view::create(input);
+
+   // from device. unique_ptr which gets automatically cleaned up when we leave.   
+   auto device_out = mutable_column_device_view::create(in_out);
+
+    // invoke the actual kernel.  
+   cudf::experimental::type_dispatcher(input.type(), 
+                                       normalize_nans_and_zeros_kernel_forwarder{},
+                                       *device_in,
+                                       *device_out,
+                                       stream);
+}
+
+}  // namespace detail
+
+/**
```

Comment:
Don't repeat Doxygen blocks in the implementation. You can duplicate them here (I usually only duplicate the brief -- less to get out of sync between header and implementation), but remove the double `**` to make sure Doxygen doesn't compile them.

```suggestion
/*
```